aggregator/election_mgr.go

```diff
@@ -728,6 +736,19 @@ func (mgr *electionManager) campaignLoop(campaignStateWatch watch.Watch) {
 			}
 			mgr.processCampaignUpdate(campaignStatus)
 		case <-mgr.doneCh:
+			electionKey := mgr.electionKey
+			// Asynchronously resign from ongoing campaign on close to avoid blocking the close
```

Comment:
Why do we want to avoid blocking the `Close` call? This seems a little to odd me because now the `Close` call can return without ensuring all the resources used by the `electionManager` are actually closed.

Reply:
Talked offline, `Close` still waits for all goroutines to exit before the call completes. It resigns asynchronously because it may take a long while to resign and this may be called while aggregator lock is held, which we don't want to happen. We simply need to ensure the ongoing campaign will be cancelled but it doesn't matter whether it's done synchronously or asynchronously.